INTEGRATION.md

```diff
@@ -0,0 +1,202 @@
+# Integration guide
+
+## Ethereum
+
+Liquid staking is achieved through `MaticX` contract and the yield-bearing ERC-20 token `MaticX` is given to the user.
+
+### 1. Stake Matic on Ethereum
+
+Send Matic and receive liquid staking MaticX token.
+_MaticX approval should be given prior._
+
+```SOLIDITY
+IMatic matic = IMatic(MATIC_ADDRESS);
+IMaticX maticX = IMaticX(MATICX_ADDRESS);
+require(matic.approve(MATICX_ADDRESS, _amountInMatic), "Not approved");
+uint256 amountInMaticX = maticX.submit(_amountInMatic);
+
+emit StakeEvent(msg.sender, msg.value, amountInMaticX);
+```
+
+### 2. Unstake Matic on Ethereum
+
+Send MaticX and create a withdrawal request.  
+_MaticX approval should be given prior._
+
+```SOLIDITY
+IMaticX maticX = IMaticX(MATICX_ADDRESS);
+require(
+    maticX.approve(MATICX_ADDRESS, _amountInMaticX),
+    "Not approved"
+);
+maticX.requestWithdraw(_amountInMaticX);
+
+emit UnstakeEvent(msg.sender, _amountInMaticX);
+```
+
+### 3. Claim Matic on Ethereum
+
+After 3-4 days (80 checkpoints), Matic can be withdrawn.
+
+```SOLIDITY
+IMatic matic = IMatic(MATIC_ADDRESS);
+IMaticX maticX = IMaticX(MATICX_ADDRESS);
+maticX.claimWithdrawal(_idx);
+uint256 amountInMatic = matic.balanceOf(msg.sender);
+
+emit ClaimEvent(msg.sender, amountInMatic);
```

Comment:
This amountInMatic should be derived from the index entry right?
Otherwise the amountInMatic will include the prior balance.